crates/uv-client/src/base_client.rs

```diff
@@ -367,6 +372,8 @@ pub struct BaseClient {
     timeout: Duration,
     /// Hosts that are trusted to use the insecure client.
     allow_insecure_host: Vec<TrustedHost>,
+    /// The number of retries to attempt on transient errors.
+    retries: u32,
```

Comment:
Just a side note: when building a retrying strategy I often like to adopt a 'deadline' approach in which an uppoer bound to the total time spent retrying is the key concept, and not so much the number of retry attempts.